server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java

```diff
@@ -63,14 +64,28 @@ public CatalogInfo addCatalog(CreateCatalog createCatalog) {
     }
   }
 
-  public ListCatalogsResponse listCatalogs() {
+  public static Integer getPageSize(Optional<Integer> maxResults) {
+    return maxResults
+            .filter(x -> (x > 0))
+            .map(x -> Math.min(x, DEFAULT_PAGE_SIZE))
+            .orElse(DEFAULT_PAGE_SIZE);
+  }
+
+  public ListCatalogsResponse listCatalogs(Optional<Integer> maxResults) {
     ListCatalogsResponse response = new ListCatalogsResponse();
     try (Session session = SESSION_FACTORY.openSession()) {
       session.setDefaultReadOnly(true);
       Transaction tx = session.beginTransaction();
       try {
+        if (maxResults.isPresent() && maxResults.get() < 0) {
+          throw new BaseException(
+                  ErrorCode.INVALID_ARGUMENT, "maxResults must be greater than or equal to 0");
+        }
+        Integer pageSize = getPageSize(maxResults);
+        Query<CatalogInfoDAO> query = session.createQuery("from CatalogInfoDAO", CatalogInfoDAO.class);
+        query.setMaxResults(pageSize);
```

Comment:
Could we hold off this change for now, we have #265 that should help in generalising this across entities.

Reply:
@vksx could you make changes to use the PagedListingHelper in this? maybe in a separate PR?